safe_transaction_service/analytics/tasks.py

```diff
@@ -9,10 +13,113 @@
 from safe_transaction_service.analytics.services.analytics_service import (
     AnalyticsService,
 )
-from safe_transaction_service.history.models import MultisigTransaction
+from safe_transaction_service.history.models import MultisigTransaction, SafeContract, SafeLastStatus
+from safe_transaction_service.history.services.balance_service import BalanceServiceProvider
 from safe_transaction_service.utils.celery import task_timeout
 from safe_transaction_service.utils.redis import get_redis
 from safe_transaction_service.utils.tasks import LOCK_TIMEOUT
+from safe_transaction_service.utils.utils import chunks
+
+logger = logging.getLogger(__name__)
+
+
+def _get_native_balance_batch(safe_addresses: List[str], balance_service) -> Tuple[int, int]:
+    """
+    Get native token balances for a batch of Safe addresses.
+    
+    :param safe_addresses: List of Safe addresses to get balances for
+    :param balance_service: BalanceService instance
+    :return: Tuple of (total_balance_wei, safes_with_balance_count)
+    """
+    batch_total_balance = 0
+    batch_safes_with_balance = 0
+    
+    for address in safe_addresses:
+        try:
+            balance_wei = balance_service.ethereum_client.get_balance(address)
```

Comment:
Good implementation overall, but I suggest leveraging the existing `BalanceService.get_balances()` method instead of direct RPC calls for native token balance checks. 
```suggestion
            balances, _ = balance_service.get_balances(
                safe_address=address,
                only_trusted=False,
                exclude_spam=False,
                limit=1  # We only need native token balance
            )
            for balance in balances:
                if balance.token_address is None:  # Native token
                    balance_wei = balance.balance
                    break
```
Benefits:
- Uses existing caching mechanisms
- Consistent with how balances are fetched elsewhere in the codebase
- Takes advantage of existing error handling and retry logic
- Will automatically benefit from any future improvements to BalanceService
- Makes future extension to include ERC20 token statistics straightforward, as the method already handles token balances

This change would make the implementation more maintainable and aligned with the service's architecture.